react/index.d.ts

```diff
@@ -73,13 +75,17 @@ declare namespace React {
     type CFactory<P, T extends Component<P, ComponentState>> = ComponentFactory<P, T>;
     type ClassicFactory<P> = CFactory<P, ClassicComponent<P, ComponentState>>;
 
-    interface DOMFactory<P extends DOMAttributes<T>, T extends Element> {
-        (props?: ClassAttributes<T> & P | null, ...children: ReactNode[]): DOMElement<P, T>;
+    interface DOMFactoryBase<P extends DOMAttributesBase<T, ChangeTarget>, T extends Element, ChangeTarget> {
+        (props?: ClassAttributes<T> & P | null, ...children: ReactNode[]): DOMElementBase<P, T, ChangeTarget>;
     }
 
-    interface HTMLFactory<T extends HTMLElement> extends DOMFactory<HTMLAttributes<T>, T> {
+    type DOMFactory<P extends DOMAttributes<T>, T extends Element> = DOMFactoryBase<P, T, EventTarget>;
+
+    interface HTMLFactoryBase<T extends HTMLElement, ChangeTarget> extends DOMFactoryBase<HTMLAttributesBase<T, ChangeTarget>, T, ChangeTarget> {
     }
 
+    type HTMLFactory<T extends HTMLElement> = HTMLFactoryBase<T, EventTarget>;
```

Comment:
Are any of the above changes necessary? We seem to have added a lot of complexity to these typings without much benefit.